docs/tutorials/07_vpp-connection.md

```diff
@@ -0,0 +1,360 @@
+```go
+func (p *HelloWorld) syncVppCall() {
+	request := &interfaces.CreateLoopback{}
+}
+```
+
+The request struct is from the generated VPP API file. The request type defines a configuration item which will be created (loopback interface in this case). Majority of the request messages contain one or more fields which can be set to given value. In our example, the `CreateLoopback` has one field `MacAddress` of type `[]byte`. Filling it will help us to create a new loopback interface with the MAC address already assigned. Since the most convenient method would be to define MAC address as `string`, we prepare short helper method which transforms our `string` type MAC address to a byte array using built-in `net` package:
+```go
+func macParser(mac string) []byte {
+	hw, err := net.ParseMAC(mac)
+	if err != nil {
+		panic(err)
+	}
+	return hw
+}
+```
+
+Now we can add the MAC address to the request body:
+```go
+func (p *HelloWorld) syncVppCall() {
+	request := &interfaces.CreateLoopback{
+		MacAddress: macParser("00:00:00:00:00:01"),
+	}
+}
+```
+
+Sending the request can in practice end up successfully or in a VPP error. To learn about the state of our request, we must define reply value. The rule within the VPP API is that the message reply has the same name + `Reply` suffix:
+```go
+func (p *HelloWorld) syncVppCall() {
+	request := &interfaces.CreateLoopback{
+		MacAddress: macParser("00:00:00:00:00:01"),
+	}
+	reply := &interfaces.CreateLoopbackReply{}
+}
+```
+
+When the request and reply messages are prepared, we can make a VPP call using our VPP channel. This is done in two steps:
+- Send the request message using `SendRequest`
+- Receive reply message calling `ReceiveReply` on request context
+
+This calling can be done in a single line (do not forget to handle the error):
+```go
+func (p *HelloWorld) syncVppCall() {
+	request := &interfaces.CreateLoopback{
+		MacAddress: macParser("00:00:00:00:00:01"),
+	}
+	reply := &interfaces.CreateLoopbackReply{}
+	err := p.vppChan.SendRequest(request).ReceiveReply(reply)
+	if err != nil {
+		panic(err)
+	}
+}
+```
+
+It is important to mention, that **the `ReceiveReply` is a blocking call**. The reply is provided only when the request is fully processed in the VPP which may take some time, depending on the request (but we still talk about milliseconds). 
+
+The reply message always provides us with the reply value called `Retval`, and sometimes with other useful data. In this case, useful data are represented by the interface index generated within the VPP and used as a unique identifier for the just-created interface. The value of `Retval` should be zero if the call was without problems, or it returns some error code if the request cannot be processed for some reason.
```

Comment:
This can be removed. The `ReceiveReply` will return [VPPApiError](https://github.com/FDio/govpp/blob/master/api/vppapi_errors.go#L19) error if the returned `Retval` was not zero and it can be easily compared with these: https://github.com/FDio/govpp/blob/345201eedce4cefae4db6914067a34000a7f978f/api/vppapi_errors.go#L33-L38